src/main/java/core/basesyntax/dao/impl/CarDaoJdbcImpl.java

```diff
@@ -0,0 +1,218 @@
+package core.basesyntax.dao.impl;
+
+import core.basesyntax.dao.CarDao;
+import core.basesyntax.exception.DataProcessingException;
+import core.basesyntax.lib.Dao;
+import core.basesyntax.model.Car;
+import core.basesyntax.model.Driver;
+import core.basesyntax.model.Manufacturer;
+import core.basesyntax.util.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class CarDaoJdbcImpl implements CarDao {
+    @Override
+    public Car create(Car car) {
+        String query = "INSERT cars (model, manufacturer_id) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setString(1, car.getModel());
+            preparedStatement.setLong(2, car.getManufacturer().getId());
+            preparedStatement.executeUpdate();
+            ResultSet resultSet = preparedStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject(1, Long.class));
+                addDriversForCar(car);
+            }
+            return car;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't insert car " + car, e);
+        }
```

Comment:
```suggestion
        String query = "INSERT cars (model, manufacturer_id) VALUES (?, ?)";
        try (Connection connection = ConnectionUtil.getConnection();
                PreparedStatement preparedStatement = connection.prepareStatement(query,
                        Statement.RETURN_GENERATED_KEYS)) {
            preparedStatement.setString(1, car.getModel());
            preparedStatement.setLong(2, car.getManufacturer().getId());
            preparedStatement.executeUpdate();
            ResultSet resultSet = preparedStatement.getGeneratedKeys();
            if (resultSet.next()) {
                car.setId(resultSet.getObject(1, Long.class));
            }
        } catch (SQLException e) {
            throw new DataProcessingException("Can't insert car " + car, e);
        }
        addDriversForCar(car);
        return car;
```

don't open one connection inside of another